app/lib/cashu/secret.ts

```diff
@@ -59,29 +40,21 @@ export const isPlainSecret = (secret: ProofSecret): secret is PlainSecret => {
  * @throws Error if the secret is a NUT-10 secret with an invalid format
  */
 export const parseSecret = (secret: string): ProofSecret => {
-  let parsed: unknown;
-  try {
-    parsed = JSON.parse(secret);
-  } catch {
-    // If JSON parsing fails, assume it's a plain string secret
+  const parsed = safeJsonParse(secret);
```

Comment:
I decided to keep the type of parsed.data as unknown here because we are doing the zod validation on the data, so at this point I'm still assuming parsed.data could be an invalid `RawNut10Secret`. Does that make sense?

Reply:
it does